### PR TITLE
Set up targets and props for AOT compilation

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets [hunk 1]
```diff
@@ -550,6 +556,42 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+
+  <Target Name="_EnsureWasmAOTWorkload" Condition="'$(UsingBlazorAOTWorkloadManifest)' != 'true'">
+    <Error
+      Text="Publishing with AOT enabled requires the .NET WebAssembly AOT workload to be installed. To learn more, visit https://aka.ms/AAb4uzl."
+      Code="BLAZORSDK1002" />
+  </Target>
+
+  <Target Name="_GatherWasmFilesToAOT">
+    <ItemGroup>
+      <WasmAssembliesToBundle Include="%(ResolvedFileToPublish.FullPath)" Exclude="@(_Exclude)" Condition="%(Extension) == '.dll' AND %(FileName) != 'Microsoft.JSInterop.WebAssembly'" />
```
Comment (captainsafia) [inline src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets:591]: What's special about the M.J.W assembly?

File: src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets [hunk 2]
```diff
@@ -550,6 +556,42 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+
+    <Copy SourceFiles="@(WasmNativeAsset)" DestinationFiles="%(WasmNativeAsset.Directory)\dotnet.$(_DotNetJsVersion).js" Condition="%(FileName) == 'dotnet' AND %(Extension) == '.js'">
+      <Output TaskParameter="CopiedFiles" ItemName="_PublishAotDotnetJsFile" />
+    </Copy>
+
+    <ItemGroup>
+      <ResolvedFileToPublish Remove="$(_BlazorDotNetJSFilePath)" />
+      <ResolvedFileToPublish Remove="@(ResolvedFileToPublish)" Condition="%(FileName) == 'dotnet' AND %(Extension) == '.wasm'" />
+
+      <ResolvedFileToPublish Include="@(_PublishAotDotnetJsFile)" AssetType="native" RelativePath="%(FileName)%(Extension)" />
```
Comment (captainsafia) [inline src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets:578]: What does the "AssetType" determine here?

**Review by captainsafia — APPROVED** [review]

File: src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets (hunk 2 above)
Comment (pranavkm) [reply src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets:578]: We use it in when producing the blazor.boot.json: https://github.com/dotnet/sdk/blob/master/src/BlazorWasmSdk/Tasks/GenerateBlazorWebAssemblyBootJson.cs#L137. It's trying to immitate the dotnet.js added by default.

File: src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets (hunk 1 above)
Comment (pranavkm) [reply src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets:591]: @lewing this is still problematic, no?

**Pull request merged**